src/main/java/org/synyx/urlaubsverwaltung/calendar/DepartmentCalendarService.java

```diff
@@ -107,6 +109,35 @@ File getCalendarForDepartment(Integer departmentId, Integer personId, String sec
         return iCalService.getCalendar(title, absences);
     }
 
+    void getCalendarForDepartment(Integer departmentId, Integer personId, String secret, Locale locale, OutputStream outputStream) throws IOException {
```

Comment:
also ich hätte statt `File` einfach den Stream zurückgegeben.

File ist eh "falsch" imo, weil es in unserem Fall hier ja keine Datei sein muss. Also der Kalender muss nicht aufgehoben werden oder so auf der Platte. Er wird einmal angefragt, generiert und zurückgegeben. Kann alles In-Memory passieren.

Reply:
und woher kommt der Stream?

Reply:
alternativ gibt man das `Calendar`-Objekt einfach bis an den Controller zurück und schreibt es dort in den Outputstream um

```java
    @GetMapping("/departments/{departmentId}/persons/{personId}/calendar")
    @ResponseBody
    public void getCalendarForDepartment(Locale locale, HttpServletResponse response, @PathVariable Integer departmentId, @PathVariable Integer personId, @RequestParam String secret) {
        setContentTypeAndHeaders(response);
        try {
            Calendar calendar = departmentCalendarService.getCalendarForDepartment2(departmentId, personId, secret, locale);
            new CalendarOutputter().output(calendar, response.getOutputStream());
        } catch (IOException |IllegalArgumentException e) {
            throw new ResponseStatusException(BAD_REQUEST, "Could not generate department calendar", e);
        } catch (CalendarException e) {
            throw new ResponseStatusException(NO_CONTENT);
        }
    }
```

Reply:
@bseber wdyt?

Reply:
Beide Lösungen setzen dann am Ende voraus, dass man mit einem `OutputStream` als Konsument zurechtkommt. Im Fall von den Controller-Methoden klappt das. Wir dürfen nur den Mail-Anhang nicht vergessen. Da ist dann noch ein bisschen was umzustellen was das Handlng von Attachements angeht: http://github.com/synyx/urlaubsverwaltung/blob/master/src/main/java/org/synyx/urlaubsverwaltung/mail/MailSenderService.java#L75

Reply:
Wir verfolgend die Idee das Thema über `Calendar` abzubilden und im `ICalViewController` / `MailServiceImpl` abzubilden